packages/ember-runtime/lib/mixins/promise_proxy.js

```diff
@@ -19,16 +19,20 @@ function tap(proxy, promise) {
   });
 
   return promise.then(value => {
-    setProperties(proxy, {
-      content: value,
-      isFulfilled: true
-    });
+    if (!get(proxy, 'isDestroyed') && !get(proxy, 'isDestroying')) {
```

Comment:
Fixing the `get` issue. Avoid the early return, not a big fan since that would mean repeating code (`return value` or `throw reason`). As an example, the second parameter for `then` would be:

```js
  (reason) => {
    if (!proxy.isDestroyed && !proxy.isDestroying) {
      setProperties(proxy, {
        reason: reason,
        isRejected: true
      });
    }
    throw reason;
  }
```

only addressing the `get` (to be pushed as soon as I check the tests) and using an early return would be:

```js
  (value) => {
    if (proxy.isDestroyed || proxy.isDestroying) {
      throw reason;
    }
    setProperties(proxy, {
      reason: reason,
      isRejected: true
    });
    throw reason;
  }
```

IMHO, it is much harder to read having a method with two paths throwing the same error but having a conditional about one of them. I can change it too, just wanted to make sure is not a simple early return.